v2/internal/controllers/crd_networking_applicationgateway_test.go

```diff
@@ -0,0 +1,389 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"log"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	network2020 "github.com/Azure/azure-service-operator/v2/api/network/v1api20201101"
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1api20220701"
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1api20200601"
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+func Test_Networking_ApplicationGateway_HTTPS_Listener_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+	appGatewayName := "appgatewaytest"
+	vnet := newVNet(tc, testcommon.AsOwner(rg), []string{"10.0.0.0/8"})
+	subnet := &network2020.VirtualNetworksSubnet{
+		ObjectMeta: tc.MakeObjectMeta("subnet"),
+		Spec: network2020.VirtualNetworks_Subnet_Spec{
+			Owner:         testcommon.AsOwner(vnet),
+			AddressPrefix: to.Ptr("10.0.0.0/24"),
+			AzureName:     "AzureAppGtwSubnet", // The subnet must have this name in Azure for Bastion to work
+		},
+	}
+	publicIPAddress := newPublicIp(tc, testcommon.AsOwner(rg))
+	tc.CreateResourceAndWait(publicIPAddress)
+	tc.CreateResourceAndWait(vnet)
+	tc.Expect(vnet.Status.Id).ToNot(BeNil())
+	tc.CreateResourceAndWait(subnet)
+	tc.Expect(subnet.Status.Id).ToNot(BeNil())
+	appGtsListnerName := "app-gw-http-listner-1"
+	httpListnerParams := map[string]string{
+		"param-key1": "applicationGateways",
+		"param-val1": appGatewayName,
+		"param-key2": "httpListeners",
+		"param-val2": appGtsListnerName,
+	}
+	appGtwFHttpListnerID, err := getFrontendPortsARMID(tc, rg, httpListnerParams)
```

Comment:
minor: You could have this function take 4 args instead of a map of string-string w/ what must be 4 values.

Might be easier to use it then?